### DISCUSSION TITLE
Starstruck badge inconsistent with other badges

### DISCUSSION THREAD

**Discussion by boazsegev** [community/community#49014] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

The Starstruck badge behaves differently from other badges, causing developer confusion and misleading profile information.

Proof can also be seen by the many questions on the feedback discussion board that seemingly arise from this issue.

### Summery

Other budges work on a **per-repo** basis and count the times they were given and for which repo they were given.

i.e., the Pull Shark budge counts the accepted PRs and shows where they were accepted.

The Starstruck badge, on the other hand, is a **per-profile** badge only dressed as a **per-repo** badge.

Instead of showing how many repos have been highly stared (expected behavior), the Starstruck badge calculates a total ranking scheme based on a `max` stared repo linked with the profile (unexpected and somewhat obscure behavior).

### For example

My account has 3 relevant repos:

- [facil.io](https://github.com/boazsegev/facil.io) with more than 1.6K stars;
- [iodine](https://github.com/boazsegev/iodine) with more than 800 stars; and
- [combine_pdf](https://github.com/boazsegev/combine_pdf) with more than 600 stars.

### My expectation - an expectation based on how other badges work - is:

- The Starstruck badge will show an "x3" (which it does, incidentally, but not for the expected reasons).
- The Starstruck badge will **list each of the relevant repos**.
- Each repo will **show the numbers of stars actually achieved** (current data).

### What actually happens:

- An x3 marker shows (but for a different reason than expected).
- Only **2 repos are listed**, using 3 listings (one repo is listed twice).
- Each listing shows **obsolete data** with the number of stars the repo had when the badge was awarded.

This isn't how other badges do it and this is why it is confusing and misleading.

A person reading the information on my profile might assume these repos aren't as successful as their star count is obsolete.

**Why a Bug?** - since this is unexpected behavior, I listed this as a bug. I understand the bug / issue is in the design rather than the implementation, but I believe that doesn't make it less of a bug.

I hope this helps improve the product.

**Comment by Anmol-Baranwal** [comment]
Starstruck badge is awarded in such a way, it lists the first repository with certain stars:

> This is awarded when any repo of yours get 16 stars.

<img src="https://user-images.githubusercontent.com/74038190/224118961-d70c44d6-bfa0-42b4-a788-dd23da7401e3.png" width="60" /> 

Similary, this is shown for 128 stars (Starstruck x2)
<img src="https://user-images.githubusercontent.com/74038190/224119190-0819e2cb-88b9-4ba6-b696-88e0e52f3631.png" width="60" /> 

So, it just lists the first repo to get 128 stars.

>There is no bug and is working as intended. 

Although, you can give feedback and suggest them to change the working.

**Reply by boazsegev** [reply]
Hi @Anmol-Baranwal ,

Thank you for replying. Perhaps I wasn't clear about the issue.

**I am aware of the design and how it works** (see `why a bug?` at the end of my post).

However, the designed and "expected" behavior (expected by the designer, that is) is unexpected by the developers and the community. Not only does **the design breaks expectations established by other badges on GitHub** (as I detailed in my post), is also breaks away from anything else the community uses - such as Badges on StackOverflow.

**In all badges we expect to see current, updated information. Only in this badge we see stale information that was true way back when the badge was awarded** (the number of stars a project **had** when the badge was given, rather than the number of stars a project **currently has**).

Another example of unexpected behavior:

**In all badges we expect multiplication to be a sign that the same badge was given again, only in this badge the multiplication sign is given for a different achievement** (silver vs. bronze).

Breaking established expectations is a bug - even if the source of the bug is in the feature's design rather than its implementation.

Thank you kindly,
  Bo.

P.S.

> Although, you can give feedback and suggest them to change the working

Yes, this exactly what I am doing here... is this not the place for this feedback?

**Reply by Anmol-Baranwal** [reply]
You should have included the tag `product feedback` according to me. Still, no worries. I have upvoted for the same.

**Comment by soodoku** [comment]
the badges don't work (update, etc.) and haven't for years. like the true state for me is not: https://github.com/soodoku
